pkg/services/apiserver/restconfig.go

```diff
@@ -0,0 +1,86 @@
+package apiserver
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	clientrest "k8s.io/client-go/rest"
+)
+
+type RestConfigProvider interface {
+	GetRestConfig(context.Context) (*clientrest.Config, error)
+}
+
+type RestConfigProviderFunc func(context.Context) (*clientrest.Config, error)
+
+func (f RestConfigProviderFunc) GetRestConfig(ctx context.Context) (*clientrest.Config, error) {
+	return f(ctx)
+}
+
+// WithoutRestConfig is a RestConfigProvider that always returns an error.
+// This is intended for use in unit tests where the rest config is not needed.
+var WithoutRestConfig = RestConfigProviderFunc(func(context.Context) (*clientrest.Config, error) {
```

Comment:
i'm not sure if there's a common pattern for this, but i was wondering if this should be named differently or in a different package if it's for use with tests? `WithoutRestConfig` feels like a confusing name to me given the use case, but not a blocker and i don't have a better idea

Reply:
This is one I'm pretty conflicted on, too... it's an ugly pattern that I wish we could somehow limit to _only_ tests (like how Rust has `#[cfg(test)]`). So to make it fit a little better & be not so terrible, I moved the interfaces and whatnot to their own file 😅 

Shouldn't be all too difficult to move it if it turns out it doesn't fit well

Reply:
I think one thing the Go codebase does is make this variable unexported, create a `_test` file with the same package name, and re-exporting it so only test files pick it up?